lint-rules.md

```diff
@@ -0,0 +1,1281 @@
+        Int ncpu = 8
+        Int modify_disk_size_gb = 0
+    }
+```
+
+### `disallowed_input_name`
+
+Any input name matching these regular expressions will be flagged: `/^[iI]n[A-Z_]/` or `/^input/i`. It is redundant and needlessly verbose to use an input's name to specify that it is an input. Input names should be short yet descriptive. Prefixing a name with `in` or `input` adds length to the name without adding clarity or context.
+
+**Group**: `naming`
+
+#### Example
+
+All the below input names would be flagged:
+
+```wdl
+    input {
+        File inBam  # This name will be flagged twice, once for the 'in' prefix, and once for being camalCase when it should be snake_case
+        File in_bam
+        File input_gtf
+    }
+```
+
+### `disallowed_output_name`
+
+Any output name matching these regular expressions will be flagged: `/^[oO]ut[A-Z_]/`, `/^output/i` or `/^..?$/`. It is redundant and needlessly verbose to use an output's name to specify that it is an output. Output names should be short yet descriptive. Prefixing a name with `out` or `output` adds length to the name without adding clarity or context. Additionally, names with only 2 characters can lead to confusion and obfuscates the content of an output. Output names should be at least 3 characters long.
+
+**Group**: `naming`
+
+#### Example
+
+All the below input names would be flagged:
+
+```wdl
+    input {
+        File outBam  # This name will be flagged twice, once for the 'out' prefix, and once for being camalCase when it should be snake_case
+        File out_bam
+        File output_gtf
+        String af
+    }
+```
+
+### `no_curly_commands`
+
+`command` blocks should be wrapped with arrows (`<<<` `>>>`) instead of curly brackets (`{` `}`). Certain Bash constructions cause problems with parsing the bracket notation. There are no such problems with the arrow notation.
+
+#### Example
+
+Good:
+
+```wdl
+    command <<<
+        echo "Hello, World!"
+    >>>
+```
+
+Bad:
+
+```wdl
+    command {
+        echo "Hello, World!"
+    }
+```
+
+### `inpersistent_container`  && `persistent_container_not_tagged`
+
+All tasks should run in a persistent container. This ensures reproducibility across time and environments. `wdl-grammar` and `wdl-ast` will look for a `sha sum` tag in your `container` declarations and warn if one is missing. A `sha` digest will always point to the exact same image, whereas tags are mutable. This mutability makes even versioned tags problematic when we want to ensure reproducibility.
+
+While the confidence in persistence gained by using `sha` digests for pulling container images is valuable, it comes at a cost. Which is a lack of human readability. So we enforce preceeding `container` entries with a comment which gives a human readable name to the image being pulled. It is imperative that the `sha` digest and the human readable tag are kept in sync. It's extra overhead compared to just using a tag directly, but we consider it well worth it.
```

Comment:
If we keep the rule, it needs to be word-smithed. For example, "Which is a lack of human readability." is a sentence fragment.